app/controllers/dish_ingredients_controller.rb

```diff
@@ -0,0 +1,12 @@
+class DishIngredientsController < ApplicationController
+
+  def new
+    DishIngredient.create(dish_id: params[:id], ingredient_id: params[:add_ingredient])
+    redirect_to "/dishes/#{params[:id]}"
+  end
+
+  def destroy
+    DishIngredient.where(dish_id: params[:id], ingredient_id: params[:ingredient_id]).first.destroy
```

Comment:
I think if you used `::find` over `::where`, then you can omit having to do a `#first` at the end. But good job getting this to work.